tests/org.eclipse.fordiac.ide.test.library/src/org/eclipse/fordiac/ide/test/library/LibraryImportTest.java

```diff
@@ -0,0 +1,298 @@
+				true, false);
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test04-1.0.0"), //$NON-NLS-1$
+				true, true);
+		LibraryManager.INSTANCE.getJobGroup().join(5000, null);
+
+		final var manifest = ManifestHelper.getContainerManifest(project);
+		dependencyAssert(manifest, "test03", "1.0.0", 0); //$NON-NLS-1$//$NON-NLS-2$
+		dependencyAssert(manifest, "test04", "1.0.0", 1); //$NON-NLS-1$//$NON-NLS-2$
+		assertEquals(2, manifest.getDependencies().getRequired().size());
+		libraryAssert("test01", "1.1.0"); //$NON-NLS-1$//$NON-NLS-2$
+	}
+
+	@Test
+	void testRangeOverlap2() throws Exception {
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test03-1.1.0"), //$NON-NLS-1$
+				true, false);
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test04-1.0.0"), //$NON-NLS-1$
+				true, true);
+		LibraryManager.INSTANCE.getJobGroup().join(5000, null);
+
+		final var manifest = ManifestHelper.getContainerManifest(project);
+		dependencyAssert(manifest, "test03", "1.1.0", 0); //$NON-NLS-1$//$NON-NLS-2$
+		dependencyAssert(manifest, "test04", "1.0.0", 1); //$NON-NLS-1$//$NON-NLS-2$
+		assertEquals(2, manifest.getDependencies().getRequired().size());
+		libraryAssert("test01", "1.5.0"); //$NON-NLS-1$//$NON-NLS-2$
+	}
+
+	@Test
+	void testRangeOverlapError() throws Exception {
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test03-1.1.0"), //$NON-NLS-1$
+				true, false);
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test04-1.1.0"), //$NON-NLS-1$
+				true, true);
+		LibraryManager.INSTANCE.getJobGroup().join(5000, null);
+
+		final var manifest = ManifestHelper.getContainerManifest(project);
+		dependencyAssert(manifest, "test03", "1.1.0", 0); //$NON-NLS-1$//$NON-NLS-2$
+		dependencyAssert(manifest, "test04", "1.1.0", 1); //$NON-NLS-1$//$NON-NLS-2$
+		assertEquals(2, manifest.getDependencies().getRequired().size());
+
+		assertFalse(project.getFolder(LibraryManager.TYPE_LIB_FOLDER_NAME).getFolder("test01").exists()); //$NON-NLS-1$
+
+		LibraryManager.INSTANCE.getJobGroup().join(1000, null);
+		FordiacMarkerHelper.JOB_GROUP.join(1000, null);
+
+		final var manifestFile = project.getFile(LibraryManager.MANIFEST);
+		final var markers = manifestFile.findMarkers(FordiacErrorMarker.LIBRARY_MARKER, false,
+				IResource.DEPTH_INFINITE);
+
+		assertTrue(markers.length > 0);
+		// markers[2].getAttribute(IMarker.MESSAGE);
+	}
+
+	@Test
+	void testDependencyChain() throws Exception {
+		LibraryManager.INSTANCE.importLibrary(project, null, java.net.URI.create("WORKSPACE_LOC/.lib/test06-1.0.0"), //$NON-NLS-1$
+				true, true);
+		LibraryManager.INSTANCE.getJobGroup().join(5000, null);
+
+		final var manifest = ManifestHelper.getContainerManifest(project);
+		dependencyAssert(manifest, "test06", "1.0.0", 0); //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
what is depedencyAssert ?
Please use a better naming syntax.
something like:
assertDependecyExistInManifest , assertDependencyMissing, etc...